### ISSUE TITLE
Trees

### ISSUE THREAD

**Issue by vifirsanova** [olyadiya/Intro_to_ML#6]
Работа в целом хорошая, но содержит одну критическую ошибку в секции AdaBoost.

| Критерий | Баллы (Макс.) | Комментарий |
|---|---|---|
| **Заполнены 7 таблиц** | 2 (3) | 6 таблиц верно (Decision Tree, Random Forest, Gradient Boosting, XGBoost, LightGBM, CatBoost). Таблица AdaBoost — неверна (см. ниже). |
| **Итоговая таблица** | 1 (1) | Заполнена, но для AdaBoost указан `learning_rate=0.5` как лучший — это следствие ошибки, реально перебор не выполнился. |
| **Ответы на вопросы** | 2 (2) | Хорошие развёрнутые ответы. |
| **Воспроизводимость** | 1 (1) | `random_state=42` везде есть. |
| **Качество кода** | 2 (3) | Минус 1.5 балла за copy-paste ошибку в AdaBoost (сломала весь эксперимент) и за дублирование вывода. |
| **Итого** | **8/10** | |

---

## Ошибка в коде (секция 3.4 AdaBoost)

**Сломанные строчки (с 6 по 9 в ячейке):**

```python
for ada in [0.5, 1.0, 1.5, 2.0]:          # переменная цикла "ada"
    start = time.time()
    clf = AdaBoostClassifier(..., learning_rate=lr, ...)   # БАГ: используется "lr" (не определена)
    ...
    results_ada.append({'learning_rate': lr, ...})         # БАГ: опять "lr"
    print(f"learning_rate={lr}: ...")                      # БАГ: опять "lr"
```

В цикле перебираются значения в переменной `ada`, но внутри используется `lr` — это переменная из предыдущего цикла (Gradient Boosting), которая осталась в памяти. В результате все 4 итерации обучились с `learning_rate=0.5` (последнее значение `lr` из предыдущего цикла). Поэтому в выводе 4 одинаковые строки с `learning_rate=0.5`.

**Как исправить:**
```python
for ada in [0.5, 1.0, 1.5, 2.0]:
    ...
    clf = AdaBoostClassifier(..., learning_rate=ada, ...)
    results_ada.append({'learning_rate': ada, ...})
    print(f"learning_rate={ada}: ...")
```

---

Работа заслуживает **8 баллов**.
